src/map/script.c

```diff
@@ -13546,6 +13575,176 @@ BUILDIN(getiteminfo)
 	return true;
 }
 
+/**
+ * Returns the value of the current equipment being parsed using static variables -
+ * current_equip_item_index and current_equip_option_index.
+ * !!Designed to be used with item_options.conf only!!
+ * *getequippedoptioninfo(<info_type>);
+ *
+ * @param (int) Types -
+ *        IT_OPT_INDEX     ID of the item option.
+ *        IT_OPT_VALUE     Amount of the bonus to be added.
+ * @return value of the type or -1.
+ */
+BUILDIN(getequippedoptioninfo)
+{
+	int val = 0, type = script_getnum(st, 2);
+	struct map_session_data *sd = NULL;
+	
+	if ((sd = script->rid2sd(st)) == NULL || status->current_equip_item_index == -1
+		|| !sd->status.inventory[status->current_equip_item_index].option[status->current_equip_option_index].index) {
+		script_pushint(st, -1);
+		return false;
+	}
+	
+	switch (type) {
+	case IT_OPT_INDEX:
+		val = sd->status.inventory[status->current_equip_item_index].option[status->current_equip_option_index].index;
+		break;
+	case IT_OPT_VALUE:
+		val = sd->status.inventory[status->current_equip_item_index].option[status->current_equip_option_index].value;
+		break;
+	default:
+		ShowError("buildin_getequippedoptioninfo: Invalid option data type %d (Max %d).\n", type, IT_OPT_MAX-1);
+		script_pushint(st, -1);
+		return false;
+	}
+	
+	script_pushint(st, val);
+	
+	return true;
+}
+
+/**
+ * Gets the option information of an equipment.
+ * *getequipoptioninfo(<equip_index>,<slot>,<type>);
+ *
+ * @param equip_index as the Index of the Equipment.
+ * @param slot        as the slot# of the Item Option (1 to MAX_ITEM_OPTIONS)
+ * @param type        IT_OPT_INDEX or IT_OPT_VALUE.
+ * @return (int) value or -1 on failure.
+ */
+BUILDIN(getequipoption)
+{
+	int val = 0, equip_index = script_getnum(st, 2);
+	int slot = script_getnum(st, 3);
+	int opt_type = script_getnum(st, 4);
+	int i = -1;
+	struct map_session_data *sd = script->rid2sd(st);
+	
+	if (sd == NULL) {
```

Comment:
Can I ask to break this elseif chain? The last check has especially nothing to do with the rest...